app/src/org/commcare/models/database/app/AppDatabaseUpgrader.java

```diff
@@ -272,48 +299,87 @@ private void safeCloseCursor(Cursor cursor) {
         }
     }
 
-    private void upgradeXFormAndroidInstallerV1(SQLiteDatabase db) {
-        // Get Global Resource Storage using AndroidPrototypeFactoryV1
-        SqlStorage<Resource> oldGlobalResourceStorage = new SqlStorage<>(
-                "GLOBAL_RESOURCE_TABLE",
-                Resource.class,
-                new ConcreteAndroidDbHelper(context, db) {
-                    @Override
-                    public PrototypeFactory getPrototypeFactory() {
-                        return AndroidPrototypeFactoryV1.getAndroidPrototypeFactoryV1(c);
-                    }
-                });
-
-        Vector<Resource> updateResourceList = new Vector<>();
-
-        // If Resource Installer is of Type XFormAndroidInstallerV1 , update it to XFormAndroidInstaller
-        // and add resource record to the updateResourceList
-        for (Resource resource : oldGlobalResourceStorage) {
-            if (resource.getInstaller() instanceof XFormAndroidInstallerV1) {
-                XFormAndroidInstallerV1 oldInstaller = (XFormAndroidInstallerV1)resource.getInstaller();
-                String contentUri = oldInstaller.getContentUri();
-                int formDefId = -1;
-                if (!StringUtils.isEmpty(contentUri)) {
-                    formDefId = Integer.valueOf(Uri.parse(contentUri).getLastPathSegment());
+    /**
+     * There can be 2 different configurations for resource table here
+     * 1. Either the v8-v9 upgrade failed and rsources are in v8 state
+     * 2. v8-v9 upgrade was successful and resources are already in v9/v10 state (Resources downloaded after the update)
+     *
+     * So we wanna assume 1 and try updating the resources to v10,
+     * if above fails, we are checking for 2 i.e resources be in v10 state.
+     * If they are not we are going to wipe the tables
+     *
+     * @param tableName Resource table that need to be upgraded
+     * @param db        App DB
+     */
+    private void upgradeToResourcesV10(String tableName, SQLiteDatabase db) {
+        try {
```

Comment:
This might sound insane, but I think it would be worth adding an assertion that kicks out of this method if tableName is the global resource table. It kind of has an innocuous name and the possibility that it fully wipes the table seems dangerous if it's the global resource table.